scalameta/semanticdb-scalac/src/main/scala/scala/meta/interactive/InteractiveSemanticdb.scala

```diff
@@ -0,0 +1,112 @@
+package scala.meta.interactive
+
+import java.io.File
+import java.net.URLClassLoader
+import scala.meta.internal.semanticdb.DatabaseOps
+import scala.reflect.io.VirtualDirectory
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interactive.Global
+import scala.tools.nsc.interactive.Response
+import scala.tools.nsc.reporters.StoreReporter
+import org.langmeta.semanticdb.Document
+
+object InteractiveSemanticdb {
+
+  def newCompiler(): Global =
+    newCompiler(thisClasspath, Nil)
+  def newCompiler(scalacOptions: List[String]): Global =
+    newCompiler(thisClasspath, scalacOptions)
+
+  /** Construct new presentation compiler with given classpath and scalac flags. */
+  def newCompiler(classpath: String, scalacOptions: List[String]): Global = {
+    val vd = new VirtualDirectory("(memory)", None)
+    val settings = new Settings
+    settings.outputDirs.setSingleOutput(vd)
+    settings.classpath.value = classpath
+    if (classpath.isEmpty) {
+      settings.usejavacp.value = true
+    }
+    settings.processArgumentString(
+      ("-Ypresentation-any-thread" :: scalacOptions).mkString(" ")
+    )
+    val compiler = new Global(settings, new StoreReporter)
+    compiler
+  }
+
+  def toDocument(compiler: Global, code: String): Document =
+    toDocument(compiler, code, "interactive.scala", 10000)
```

Comment:
How is the filename parameter used by the PC?

Reply:
the PC keeps internal state with open files etc, so you should use unique names from an editor with multiple buffers. For basic cases like testing it's nice to have a default.